### PR TITLE
Allow items with a single valued attribute to be added from listing page

### PR DESCRIPTION
**Pull request by scottcwilson**
Examples would be products with PDF documents or Zip files.

### CODE REVIEW

File: includes/functions/functions_lookups.php [hunk 1]
```diff
@@ -270,6 +270,63 @@ function zen_has_product_attributes($products_id, $not_readonly = 'true') {
...
+      }
+      // if there's any type from $noSingles, we fail
+      if (in_array($field['options_type'], $noSingles)) {
+        $fail = true;
+        break;
+      }
+    }
+
+    // return 1 to indicate selections must be made, so a more-info button needs to be presented
+    if ($fail) return 1;
```
Comment (drbyte) [inline includes/functions/functions_lookups.php]: Looking at this code now, the `$fail = true; break;` can both be replaced with just `return 1;`, and then this `if($fail)return 1` can be removed.
Maybe incorporate the comment by changing ", we fail" to "we return 1 so selections can be made".

**Comment by drbyte** [comment]
Oops - somehow you added credit card installments files 👀

**Comment by scottcwilson** [comment]
ack sorry that was for debugging.

**Comment by mc12345678** [comment]
No handling for other product types in `zen_requires_attribute_selection`?

**Comment by scottcwilson** [comment]
Your PR is welcome if you can see a miss.

**Comment by drbyte** [comment]
> No handling for other product types in `zen_requires_attribute_selection`?

It's not product-type specific anyway.

**Pull request merged**